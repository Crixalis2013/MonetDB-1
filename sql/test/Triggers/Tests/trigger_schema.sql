-- Verify the trigger's schema is the one set every time the trigger is fired
START TRANSACTION;
create schema foo;
set schema foo;
create table t1 (id serial, ref bigint);
create table t2 (id serial, ref bigint);

create function f(ref bigint) returns bigint begin return 10*ref; end;

create schema bar;
set schema bar;
create function f(ref bigint) returns bigint begin return 100*ref; end;
create table t1 (id serial, ref bigint);
create table t2 (id serial, ref bigint);

create trigger extra_insert
    AFTER INSERT ON t1 referencing new row as new_row
    FOR EACH statement insert into t2(ref) values (f(new_row.ref));

insert into t1(ref) values (10);
set schema foo;
insert into bar.t1(ref) values (10);

select * from bar.t1;
select * from bar.t2;
select * from foo.t1;
select * from foo.t2;
ROLLBACK;

START TRANSACTION;
create schema foo;
set schema foo;
create table t1 (id serial, ref bigint);
create table t2 (id serial, ref bigint);
create schema bar;
set schema bar;
create trigger foo.extra_insert
    AFTER INSERT ON t1 referencing new row as new_row
    FOR EACH statement insert into t2(ref) values (new_row.ref); --error, a trigger will be placed on its table's schema, specify the schema on the table reference, ie ON clause instead
ROLLBACK;
