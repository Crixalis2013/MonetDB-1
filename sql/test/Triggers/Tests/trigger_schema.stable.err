stderr of test 'trigger_schema` in directory 'sql/test/Triggers` itself:


# 14:26:57 >  
# 14:26:57 >  "mclient" "-lsql" "-ftest" "-tnone" "-Eutf-8" "-i" "-e" "--host=/var/tmp/mtest-75225" "--port=38787"
# 14:26:57 >  

MAPI  = (monetdb) /var/tmp/mtest-78050/.s.monetdb.34605
QUERY = create trigger foo.extra_insert
            AFTER INSERT ON t1 referencing new row as new_row
            FOR EACH statement insert into t2(ref) values (new_row.ref); --error, a trigger will be placed on its table's schema, specify the schema on the table reference, ie ON clause instead
ERROR = !CREATE TRIGGER: a trigger will be placed on the respective table's schema, specify the schema on the table reference, ie ON clause instead
CODE  = 42000

# 14:26:57 >  
# 14:26:57 >  "Done."
# 14:26:57 >  

