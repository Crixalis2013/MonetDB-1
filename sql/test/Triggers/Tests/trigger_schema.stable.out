stdout of test 'trigger_schema` in directory 'sql/test/Triggers` itself:


# 14:26:57 >  
# 14:26:57 >  "mclient" "-lsql" "-ftest" "-tnone" "-Eutf-8" "-i" "-e" "--host=/var/tmp/mtest-75225" "--port=38787"
# 14:26:57 >  

#START TRANSACTION;
#create schema foo;
#set schema foo;
#create table t1 (id serial, ref bigint);
#create table t2 (id serial, ref bigint);
#create function f(ref bigint) returns bigint begin return 10*ref; end;
#create schema bar;
#set schema bar;
#create function f(ref bigint) returns bigint begin return 100*ref; end;
#create table t1 (id serial, ref bigint);
#create table t2 (id serial, ref bigint);
#create trigger extra_insert
#    AFTER INSERT ON t1 referencing new row as new_row
#    FOR EACH statement insert into t2(ref) values (f(new_row.ref));
#insert into t1(ref) values (10);
[ 1	]
#set schema foo;
#insert into bar.t1(ref) values (10);
[ 1	]
#select * from bar.t1;
% bar.t1,	bar.t1 # table_name
% id,	ref # name
% int,	bigint # type
% 1,	2 # length
[ 1,	10	]
[ 2,	10	]
#select * from bar.t2;
% bar.t2,	bar.t2 # table_name
% id,	ref # name
% int,	bigint # type
% 1,	4 # length
[ 1,	1000	]
[ 2,	1000	]
#select * from foo.t1;
% foo.t1,	foo.t1 # table_name
% id,	ref # name
% int,	bigint # type
% 1,	1 # length
#select * from foo.t2;
% foo.t2,	foo.t2 # table_name
% id,	ref # name
% int,	bigint # type
% 1,	1 # length
#ROLLBACK;

# 14:26:57 >  
# 14:26:57 >  "Done."
# 14:26:57 >  

